### PR TITLE
Pull out timing observations, more jsonable arguments

### LINKED ISSUE #434: Too-slow health check doesn't say what was too slow
From https://travis-ci.org/LeastAuthority/txkube/builds/194934837

```
Traceback (most recent call last):
  File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/txkube/test/test_model.py", line 94, in test_roundtrip
    def test_roundtrip(self, obj):
  File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/hypothesis/core.py", line 438, in wrapped_test
    HealthCheck.too_slow,
  File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/hypothesis/core.py", line 306, in fail_health_check
    raise FailedHealthCheck(message)
hypothesis.errors.FailedHealthCheck: Data generation is extremely slow: Only produced 9 valid examples in 1.13 seconds (0 invalid ones and 1 exceeded maximum size). Try decreasing size of the data you're generating (with e.g.average_size or max_leaves parameters).
See https://hypothesis.readthedocs.io/en/latest/healthchecks.html for more information about this. If you want to disable just this health check, add HealthCheck.too_slow to the suppress_health_check settings for this test.
```

Fortunately there's only a single strategy used by the test that failed here so (by reading the source) it's easy to find out what it was.  If there had been multiple strategies, it would have been more of a pain to track down.

### PR DESCRIPTION
**Pull request by Zac-HD**
Plus a note in the docs inspired by discussion on https://github.com/tyche-pbt/tyche-extension/issues/3.  

Cc @hgoldstein95 for timing metadata per your email.

- [x] Report `generate:{argname}` separately for each argument (and interactive draw).  If we update the too_slow healthcheck message to include observability info (or suggest turning it on), this will fix #434!

### CODE REVIEW

**Comment by Zac-HD** [comment]
Hmm.  With a bit of work on the internals, we could instead report the draw time _for each argument_, corresponding to the entries in the `arguments` mapping (i.e. each argument to `@given()` and each interactive draw from `st.data()`, plus a residual from e.g. `st.functions()`).  @hgoldstein95, do you think that'd be useful?

File: hypothesis-python/docs/schema_observations.json [hunk 1]
```diff
@@ -44,6 +44,14 @@
                         "uniqueItems": true
                     }
                 },
+                "timing": {
+                    "type": "object",
+                    "description": "The time in seconds taken by non-overlapping parts of this test case.  Hypothesis reports generate_before_test, execute_test, and generate_during_test.",
```
Comment (tybug) [inline hypothesis-python/docs/schema_observations.json:49]: what would be an example of an "overlapping" part of the test case? Is this just intended to convey that the three phases — before, execute, during — are disjoint? I'm left a bit confused (which means that the casual reader may be as well).

**Review by tybug — COMMENTED** [review]
pulling out timing data into its own key makes a lot of sense, and reporting per-argument draw times would be even better! In a perfect world I would love to observe the timings of every component strategy draw as well 🙂 (I mentioned as much to @hgoldstein95 in our interview).

File: hypothesis-python/docs/schema_observations.json (hunk 1 above)
Comment (Zac-HD) [reply hypothesis-python/docs/schema_observations.json:49]: Yep, intended to express that these are disjoint times such that taking the sum gives the correct total.  (though test execution and generate-during are interleaved, we subtract the latter from our raw measurement to yield the former)

**Comment by Zac-HD** [comment]
> pulling out timing data into its own key makes a lot of sense, and reporting per-argument draw times would be even better! In a perfect world I would love to observe the timings of every component strategy draw as well 🙂

Component strategy draws are hard, because we don't have human-readable labels.  We could add compute and collect those, but we'd then have the problem that _most_ component strategy draws are internals rather than a meaningful component in the user-facing API/mental model.  I remain pretty nervous about introducing too much complexity, and improving life for maintainers and power users at the expense of novices.

Splitting out per-argument timing is definitely doable though, and _that_ is certainly easy to interpret.  Keys `execute_test`, and `generate:{argname}` for each argname key in the `arguments` mapping 😁

File: hypothesis-python/tests/conjecture/test_engine.py [hunk 2]
```diff
@@ -1546,3 +1547,30 @@ def test(data):
...
+      count | fraction |    slowest draws (seconds)
+  a |    3  |     65%  |      --      --      --   0.357,  2.000
+  b |    8  |     16%  |   0.100,  0.100,  0.100,  0.111,  0.123
+  c |    3  |      8%  |      --      --   0.030,  0.050,  0.200
+  i |    2  |      2%  |      --      --      --      --   0.090
+  h |    1  |      2%  |      --      --      --      --   0.080
+  (skipped 4 rows of fast draws)"""
+    got = state.timing_report()
+    print(got)
+    assert expected == got
```
Comment (tybug) [inline hypothesis-python/tests/conjecture/test_engine.py:1576]: personally, I'd expect this to read left to right: slowest draws on the left, and fastest on the right. I have no scientific backing for this besides an opinion, though.

**Pull request merged**